twitter_pr.rs

```diff
@@ -0,0 +1,96 @@
+extern crate reqwest;
+
+use std::sync::mpsc::{Receiver, channel};
+use serde::{Serialize, Deserialize};
+use tokio::stream::StreamExt;
+use tokio::io::{AsyncBufReadExt, StreamReader};
+use serde_json::{Deserializer, StreamDeserializer};
+use std::io::Write;
+
+pub struct TwitterApi {
+    pub bearer_api_token: String
+}
+
+struct RequestStream {
+    receiver: Receiver<bytes::Bytes>
+}
+
+impl std::io::Read for RequestStream {
+    fn read(&mut self, mut buf: &mut [u8]) -> std::io::Result<usize> {
+        match self.receiver.recv() {
+            Ok(next) => {
+                let vec = next.to_vec();
+                let vec_len = vec.len();
+                println!("Incoming -> {}", String::from_utf8(vec.clone()).unwrap_or("err".to_string()));
+                buf.write_all(vec.as_slice());
+                return Ok(vec_len);
+            }
+            Err(_e) => panic!("error!")
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Tweet {
+    pub text: String,
+}
+
+#[derive(Deserialize)]
+struct Wrapper<T> {
+    data: T,
+}
+
+impl TwitterApi {
+    pub fn new(api_token: String) -> TwitterApi {
+        TwitterApi {
+            bearer_api_token: api_token
+        }
+    }
+
+    pub async fn listen_for_tweets(&self) -> Receiver<Tweet> {
+        let (sender, receiver) = channel::<Tweet>();
```

Comment:
I think these are unused. Is that right? Either rustc or `cargo clippy` should catch such cases.